robot/web/stream/camera/camera.go

```diff
@@ -19,3 +24,23 @@ func Camera(robot robot.Robot, stream gostream.Stream) (camera.Camera, error) {
 	}
 	return cam, nil
 }
+
+// VideoSourceFromCamera converts a camera resource into a gostream VideoSource.
+// This is useful for streaming video from a camera resource.
+func VideoSourceFromCamera(ctx context.Context, cam camera.Camera) gostream.VideoSource {
+	reader := gostream.VideoReaderFunc(func(ctx context.Context) (image.Image, func(), error) {
+		img, err := camera.DecodeImageFromCamera(ctx, "", nil, cam)
+		if err != nil {
+			return nil, func() {}, err
+		}
+		return img, func() {}, nil
+	})
+	camProps, err := cam.Properties(ctx)
+	if err != nil {
+		camProps = camera.Properties{}
+	}
+	if camProps.IntrinsicParams == nil {
+		return gostream.NewVideoSource(reader, prop.Video{Width: 0, Height: 0})
```

Comment:
Is 0,0 a valid thing for a video stream to ever have? Are there good defaults to set this to instead?

Reply:
thoughts on pulling a frame and taking a look at its resolution like how we do with video store? @seanavery

Reply:
No overriding being done in this current code. Could we pull an image and return if its successful? Then fall back to properties if not? Then error if neither work?

Reply:
That sounds good, it is also worth it to add a test since this function is a candidate for the glue to web_server.

Reply:
Since this helper will be primarily used to create video sources for the stream server streams, it is not that big of a deal to not know the width and height. The processInputFrames routine will check width and height of incoming frames and create a new encoder instance if there is a mismatch. See [here](https://github.com/viamrobotics/rdk/blob/main/gostream/stream.go#L271-L283).

Reply:
Added test file and test.

Changed logic to not care about video prop if fail